src/legacy/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -235,19 +242,8 @@ export class EmbeddedVisualizeHandler {
       });
     }
 
-    let fetchRequired = false;
-    if (params.hasOwnProperty('timeRange')) {
-      fetchRequired = true;
-      this.dataLoaderParams.timeRange = params.timeRange;
-    }
-    if (params.hasOwnProperty('filters')) {
-      fetchRequired = true;
-      this.dataLoaderParams.filters = params.filters;
-    }
-    if (params.hasOwnProperty('query')) {
-      fetchRequired = true;
-      this.dataLoaderParams.query = params.query;
-    }
+    const fetchRequired = ['timeRange', 'filters', 'query'].some(key => params.hasOwnProperty(key));
+    this.dataLoaderParams = { ...this.dataLoaderParams, ...otherParams };
```

Comment:
The reason this was changed is because I had originally added `'abortSignal'` in here, and didn't feel like copying/pasting the exact same code 4 times made sense when we could just do it like this. Then I ended up removing `'abortSignal'` from here and actually creating it in the handler. I don't mind reverting these changes if it makes people feel more comfortable :)